refaster-support/src/main/java/tech/picnic/errorprone/refaster/annotation/OnlineDocumentation.java

```diff
@@ -0,0 +1,49 @@
+package tech.picnic.errorprone.refaster.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Signals that a Refaster template or group of Refaster templates comes with online documentation.
+ *
+ * <p>The provided value may be a full URL, or a URL pattern containing either or both of the
+ * {@value TOP_LEVEL_CLASS_URL_PLACEHOLDER} and {@value NESTED_CLASS_URL_PLACEHOLDER} placeholders.
+ *
+ * <p>By default it is assumed that the Refaster template(s) are documented on the Error Prone
+ * Support website. Annotations on nested classes override the documentation URL associated with any
+ * enclosing class.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.SOURCE)
+public @interface OnlineDocumentation {
+  /**
+   * The URL placeholder value that will be replaced with the name of the top-level class in which
+   * the annotated Refaster template is located.
+   */
+  String TOP_LEVEL_CLASS_URL_PLACEHOLDER = "${topLevelClassName}";
+  /**
+   * The URL placeholder value that will be replaced with the name of the nested class in which the
+   * annotated Refaster template is located, if applicable.
+   *
+   * <p>If the Refaster template is not defined in a nested class then this placeholder will be
+   * replaced with the empty string. In case the Refaster template is syntactically nested inside a
+   * deeper hierarchy of classes, then this placeholder will be replaced with concatenation of the
+   * names of all these classes (except the top-level class name), separated by dots.
+   */
+  String NESTED_CLASS_URL_PLACEHOLDER = "${nestedClassName}";
+
+  /**
+   * The URL or URL pattern of the website at which the annotated Refaster template(s) are
+   * documented.
+   *
+   * @return A non-{@code null} string, optionally containing the {@value
+   *     TOP_LEVEL_CLASS_URL_PLACEHOLDER} and {@value NESTED_CLASS_URL_PLACEHOLDER} placeholders.
+   */
+  String value() default
+      "https://error-prone.picnic.tech/refastertemplates/"
```

Comment:
If we agree that they're _rules_, then we should update the website URLs before widely publicizing it...

Reply:
Having an agreement is indeed a blocker for the PR to not have to change this URL. And I agree with the idea to call the whole thing a rule and the part of it the templates (i.e. template of a piece of code that the rule matches/writes to). However, I haven't (yet) fully read the original author's [paper](https://research.google.com/pubs/archive/41876.pdf). So ultimately I will agree with what you, @Stephan202 and @rickie, agree on given your greater experience in using and developing these.

Reply:
> Edit: and I'd scope the name consistency change only for the URL for now and touched LoC, to proceed with this PR.

Indeed. In fact, I'd even be okay with deferring all renaming to a separate PR, since the current use of "template", even if we wish to change it, is quite consistent with the rest of the code base.

Reply:
Maybe that would even be the best option, deferring to a separate PR. Anyway, yes it should be changed 💪🏻.